socorro/tests/processor/test_cache_manager.py

```diff
@@ -0,0 +1,397 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
The cache manager code and its tests were pulled in from Eliot which has been running it in production for over a year now. Having said that, the Socorro minidump stackwalker saves files in a deep directory tree, but Eliot saves all its files in the same directory--no subdirectories. The code for reacting to new directories (which Eliot didn't use) was buggy.

Thus this version and its tests is just like the Eliot version, but with the support for directory trees fixed.